server/src/main/java/org/opensearch/discovery/PeerFinder.java

```diff
@@ -139,26 +139,15 @@ public PeerFinder(
         );
     }
 
-    public ActionListener<Void> nodeCommissionedListener() {
-        return new ActionListener<Void>() {
-            @Override
-            public void onResponse(Void unused) {
-                localNodeDecommissioned = false;
-                findPeersInterval = DISCOVERY_FIND_PEERS_INTERVAL_SETTING.get(settings);
-                logger.info("updated findPeersInterval to [{}] as node is commissioned", findPeersInterval);
-            }
-
-            @Override
-            public void onFailure(Exception e) {
-                localNodeDecommissioned = true;
-                findPeersInterval = DISCOVERY_FIND_PEERS_INTERVAL_DURING_DECOMMISSION_SETTING.get(settings);
-                logger.info("updated findPeersInterval to [{}] as node is decommissioned", findPeersInterval);
-            }
-        };
+    public void setFindPeersInterval(boolean localNodeCommissioned) {
+        this.localNodeCommissioned = localNodeCommissioned;
+        findPeersInterval = localNodeCommissioned ? DISCOVERY_FIND_PEERS_INTERVAL_SETTING.get(settings) :
+            DISCOVERY_FIND_PEERS_INTERVAL_DURING_DECOMMISSION_SETTING.get(settings);
+        logger.info("setting findPeersInterval to [{}] as node commission status = [{}]", findPeersInterval, localNodeCommissioned);
```

Comment:
The case where I think backoff should help is when someone accidentally decommissions and wants to recommission quickly, do you think we can create a backoff policy with a plateau

Reply:
This makes sense for a quick recommission on accidental decommission. Do you think we can take this up as follow up PR for this and this one can go independently? Open a issue for the same - #4615